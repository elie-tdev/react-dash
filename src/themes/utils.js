export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
