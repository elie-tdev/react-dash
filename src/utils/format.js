const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
