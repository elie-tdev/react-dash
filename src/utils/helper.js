const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
