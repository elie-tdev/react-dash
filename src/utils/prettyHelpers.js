export const prettyCurrency = (amount, currency = 'USD') => {
  if (amount == null) {
    return false;
  }
  if (amount < 0) {
    amount = 0;
  }
  return amount.toLocaleString('en', { style: 'currency', currency });
};

export const prettyDateTime = time => {
  if (time == null) {
    return false;
  }
  let d = new Date(time);
  return (
    d.toLocaleDateString('en', {
      month: '2-digit',
      day: '2-digit',
      year: '2-digit',
    }) +
    ' at ' +
    d.toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' }).replace(/^0(?:0:0?)?/, '')
  );
};

export const prettyDate = time => new Date(time).toLocaleDateString('en', {});

export const prettyTime = time =>
  new Date(time).toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' });

export const prettyHour = time =>
  new Date(time)
    .toLocaleTimeString('en', { hour: '2-digit', minute: '2-digit' })
    .replace(/^0(?:0:0?)?/, '');

export const prettyPhone = number =>
  number.substr(2).replace(/(\d{1,2})(\d{1})?(\d{1,3})?(\d{1,4})?/, function (_, p1, p2, p3, p4) {
    let output = '';
    if (p1) output = `${p1}`;
    if (p2) output += `${p2}-`;
    if (p3) output += `${p3}-`;
    if (p4) output += `${p4}`;
    return output;
  });

export const minimumDays = (date, numDays) => {
  const d = new Date(date);
  const totalNumDays = daysIntoYear(d) + numDays;
  return Boolean(daysIntoYear(d) >= totalNumDays || totalNumDays <= 365);
};

export const daysIntoYear = date => {
  return (
    (Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()) -
      Date.UTC(date.getFullYear(), 0, 0)) /
    24 /
    60 /
    60 /
    1000
  );
};

export const cleanAmountRegex = value => {
  return value.replace(/[^0-9]/g, '');
};

export const cleanAmount = value => {
  const pos = value.indexOf('.');
  let result;
  if (pos !== -1) {
    const part1 = value.substr(0, pos);
    const part2 = value.substr(pos + 1);
    result = cleanAmountRegex(part1) + '.' + cleanAmountRegex(part2);
  } else {
    result = cleanAmountRegex(value);
  }
  return result;
};
