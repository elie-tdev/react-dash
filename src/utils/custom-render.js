import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
