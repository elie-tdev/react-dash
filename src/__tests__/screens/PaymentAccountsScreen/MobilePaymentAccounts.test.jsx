import React from 'react';

import MobilePaymentAccounts from '@/screens/PaymentAccountsScreen/MobilePaymentAccounts';
import { fireEvent, render } from '@/utils/custom-render';

describe('MobilePaymentAccounts', () => {
  test('renders MobilePaymentAccounts', () => {
    const { container } = render(<MobilePaymentAccounts />);
    expect(container).toMatchSnapshot();
  });

  test('renders MobilePaymentAccounts with bank accounts', () => {
    const { container } = render(
      <MobilePaymentAccounts
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('onClick of New Payment Account', async () => {
    const setIsNewPaymentAccountModalOpen = jest.fn();
    const { container, getAllByTestId } = render(
      <MobilePaymentAccounts
        onClick={() => setIsNewPaymentAccountModalOpen(true)}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
        setIsNewPaymentAccountModalOpen={setIsNewPaymentAccountModalOpen}
      />,
    );
    await new Promise(resolve => setTimeout(resolve, 0));

    expect(await getAllByTestId(/PrimaryButton/i)).toHaveLength(1);
    fireEvent.click(await getAllByTestId(/PrimaryButton/i)[0]);
    await expect(setIsNewPaymentAccountModalOpen).toHaveBeenCalled();
    await expect(container).toMatchSnapshot();
  });

  test('onClick of Account list item', async () => {
    const setIsEditPaymentAccountModalOpen = jest.fn();
    const setEditingAccountIndex = jest.fn();
    const { container } = render(
      <MobilePaymentAccounts
        onClick={() => {
          setEditingAccountIndex(1);
          setIsEditPaymentAccountModalOpen(true);
        }}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: true,
          },
        ]}
        setIsEditPaymentAccountModalOpen={setIsEditPaymentAccountModalOpen}
        setEditingAccountIndex={setEditingAccountIndex}
      />,
    );
    await new Promise(resolve => setTimeout(resolve, 0));

    const icons = container.querySelector('button[class="MuiButtonBase-root MuiIconButton-root"]');
    fireEvent.click(icons);
    await expect(setEditingAccountIndex).toHaveBeenCalled();
    await expect(setIsEditPaymentAccountModalOpen).toHaveBeenCalled();
    await expect(container).toMatchSnapshot();
  });

  test('onClick of Account list item for non primary account', async () => {
    const setIsEditPaymentAccountModalOpen = jest.fn();
    const setEditingAccountIndex = jest.fn();
    const { container } = render(
      <MobilePaymentAccounts
        onClick={() => {
          setEditingAccountIndex(1);
          setIsEditPaymentAccountModalOpen(true);
        }}
        bankAccounts={[
          {
            routingNumber: '12341234',
            accountNumber: '123412341234',
            accountType: 'Checking',
            isPrimary: false,
          },
        ]}
        setIsEditPaymentAccountModalOpen={setIsEditPaymentAccountModalOpen}
        setEditingAccountIndex={setEditingAccountIndex}
      />,
    );
    await new Promise(resolve => setTimeout(resolve, 0));
    const icons = container.querySelector('button[class="MuiButtonBase-root MuiIconButton-root"]');
    fireEvent.click(icons);
    await expect(setEditingAccountIndex).toHaveBeenCalled();
    await expect(setIsEditPaymentAccountModalOpen).toHaveBeenCalled();
    await expect(container).toMatchSnapshot();
  });
});
