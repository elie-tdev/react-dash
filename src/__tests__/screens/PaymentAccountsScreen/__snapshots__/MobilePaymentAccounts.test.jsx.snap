// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobilePaymentAccounts onClick of Account list item 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-81 makeStyles-paymentAccountsBoxContainer-69"
  >
    <div
      class="MuiBox-root MuiBox-root-82 makeStyles-paymentAccountsBox-70 makeStyles-paymentAccountsBox-79"
    >
      <h1
        class="MuiTypography-root makeStyles-row-71 MuiTypography-h1"
      >
        Payment Accounts
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-72 makeStyles-row-71 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <div
        aria-label="chosenAccount"
        class="MuiFormGroup-root makeStyles-radioGroup-73"
        role="radiogroup"
      >
        <ul
          class="MuiList-root makeStyles-itemList-74 MuiList-padding"
        >
          <div
            class="MuiBox-root MuiBox-root-83 makeStyles-listPrimary-75"
          >
            <p
              class="MuiTypography-root makeStyles-defaultAccount-78 MuiTypography-body1"
            >
              Default Account
            </p>
            <li
              class="MuiListItem-container"
            >
              <div
                class="MuiListItem-root makeStyles-listItem-76 makeStyles-listItem-85 MuiListItem-gutters MuiListItem-secondaryAction"
              >
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-86 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-87 Mui-checked MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-89"
                        name="chosenAccount"
                        type="radio"
                        value="0"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-90 PrivateRadioButtonIcon-checked-92"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-91"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Checking
                     
                    •••• 
                    1234
                  </span>
                </label>
              </div>
              <div
                class="MuiListItemSecondaryAction-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </li>
          </div>
        </ul>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-93 makeStyles-root-94 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="PrimaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + New Payment Account
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`MobilePaymentAccounts onClick of Account list item for non primary account 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-107 makeStyles-paymentAccountsBoxContainer-95"
  >
    <div
      class="MuiBox-root MuiBox-root-108 makeStyles-paymentAccountsBox-96 makeStyles-paymentAccountsBox-105"
    >
      <h1
        class="MuiTypography-root makeStyles-row-97 MuiTypography-h1"
      >
        Payment Accounts
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-98 makeStyles-row-97 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <div
        aria-label="chosenAccount"
        class="MuiFormGroup-root makeStyles-radioGroup-99"
        role="radiogroup"
      >
        <ul
          class="MuiList-root makeStyles-itemList-100 MuiList-padding"
        >
          <li
            class="MuiListItem-container"
          >
            <div
              class="MuiListItem-root makeStyles-listItem-102 makeStyles-listItem-110 makeStyles-listBorder-103 MuiListItem-gutters MuiListItem-secondaryAction"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-111 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-114"
                      name="chosenAccount"
                      type="radio"
                      value="0"
                    />
                    <div
                      class="PrivateRadioButtonIcon-root-115"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-116"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Checking
                   
                  •••• 
                  1234
                </span>
              </label>
            </div>
            <div
              class="MuiListItemSecondaryAction-root"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-118 makeStyles-root-119 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="PrimaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + New Payment Account
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`MobilePaymentAccounts onClick of New Payment Account 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-55 makeStyles-paymentAccountsBoxContainer-43"
  >
    <div
      class="MuiBox-root MuiBox-root-56 makeStyles-paymentAccountsBox-44 makeStyles-paymentAccountsBox-53"
    >
      <h1
        class="MuiTypography-root makeStyles-row-45 MuiTypography-h1"
      >
        Payment Accounts
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-46 makeStyles-row-45 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <div
        aria-label="chosenAccount"
        class="MuiFormGroup-root makeStyles-radioGroup-47"
        role="radiogroup"
      >
        <ul
          class="MuiList-root makeStyles-itemList-48 MuiList-padding"
        >
          <div
            class="MuiBox-root MuiBox-root-57 makeStyles-listPrimary-49"
          >
            <p
              class="MuiTypography-root makeStyles-defaultAccount-52 MuiTypography-body1"
            >
              Default Account
            </p>
            <li
              class="MuiListItem-container"
            >
              <div
                class="MuiListItem-root makeStyles-listItem-50 makeStyles-listItem-59 MuiListItem-gutters MuiListItem-secondaryAction"
              >
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-60 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-61 Mui-checked MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-63"
                        name="chosenAccount"
                        type="radio"
                        value="0"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-64 PrivateRadioButtonIcon-checked-66"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-65"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Checking
                     
                    •••• 
                    1234
                  </span>
                </label>
              </div>
              <div
                class="MuiListItemSecondaryAction-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </li>
          </div>
        </ul>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-67 makeStyles-root-68 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="PrimaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + New Payment Account
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`MobilePaymentAccounts renders MobilePaymentAccounts 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-13 makeStyles-paymentAccountsBoxContainer-1"
  >
    <div
      class="MuiBox-root MuiBox-root-14 makeStyles-paymentAccountsBox-2 makeStyles-paymentAccountsBox-11"
    >
      <h1
        class="MuiTypography-root makeStyles-row-3 MuiTypography-h1"
      >
        Payment Accounts
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-4 makeStyles-row-3 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <div
        aria-label="chosenAccount"
        class="MuiFormGroup-root makeStyles-radioGroup-5"
        role="radiogroup"
      >
        <ul
          class="MuiList-root makeStyles-itemList-6 MuiList-padding"
        />
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-15 makeStyles-root-16 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="PrimaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + New Payment Account
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`MobilePaymentAccounts renders MobilePaymentAccounts with bank accounts 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-29 makeStyles-paymentAccountsBoxContainer-17"
  >
    <div
      class="MuiBox-root MuiBox-root-30 makeStyles-paymentAccountsBox-18 makeStyles-paymentAccountsBox-27"
    >
      <h1
        class="MuiTypography-root makeStyles-row-19 MuiTypography-h1"
      >
        Payment Accounts
      </h1>
      <p
        class="MuiTypography-root makeStyles-caption-20 makeStyles-row-19 MuiTypography-body1"
      >
        Your default bank account will be used for any future payments scheduled through the dashboard. It will not impact previously scheduled payments.
      </p>
      <div
        aria-label="chosenAccount"
        class="MuiFormGroup-root makeStyles-radioGroup-21"
        role="radiogroup"
      >
        <ul
          class="MuiList-root makeStyles-itemList-22 MuiList-padding"
        >
          <div
            class="MuiBox-root MuiBox-root-31 makeStyles-listPrimary-23"
          >
            <p
              class="MuiTypography-root makeStyles-defaultAccount-26 MuiTypography-body1"
            >
              Default Account
            </p>
            <li
              class="MuiListItem-container"
            >
              <div
                class="MuiListItem-root makeStyles-listItem-24 makeStyles-listItem-33 MuiListItem-gutters MuiListItem-secondaryAction"
              >
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-34 MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked-35 Mui-checked MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-37"
                        name="chosenAccount"
                        type="radio"
                        value="0"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-38 PrivateRadioButtonIcon-checked-40"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-39"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                  >
                    Checking
                     
                    •••• 
                    1234
                  </span>
                </label>
              </div>
              <div
                class="MuiListItemSecondaryAction-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </li>
          </div>
        </ul>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root makeStyles-root-41 makeStyles-root-42 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
      data-testid="PrimaryButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        + New Payment Account
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
